iphone/Classes/TiUIListView.m

```diff
@@ -1653,18 +1653,71 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
             [self.proxy fireEvent:@"pull" withObject:[NSDictionary dictionaryWithObjectsAndKeys:NUMBOOL(pullActive),@"active",nil] withSource:self.proxy propagate:NO reportSuccess:NO errorCode:0 message:nil];
         }
     }
+}
+
+// Gets the visible index paths and removes the top row that is not completely visible
+-(NSIndexPath*)firstVisibleIndexPath:(NSInteger)index
+{
+    UITableView* tableView = [self tableView];
+    NSArray* array = [tableView indexPathsForVisibleRows];
+    NSIndexPath* indexPath = [array objectAtIndex: index];
+    CGRect cellRect = [tableView rectForRowAtIndexPath:indexPath];
+    CGRect tableRect = [tableView bounds];
     
+    if(tableRect.origin.y <= (cellRect.origin.y - [self tableView:tableView heightForHeaderInSection:[indexPath section]])) {
+        return indexPath;
+    }
+
+    index += 1;
+    if([array count] >= index) {
+        return [self firstVisibleIndexPath:index];
+    }
+    if(indexPath == nil) {
+        return [NSIndexPath indexPathForRow:0 inSection:0];
+    }
+    return indexPath;
+}
+
+
+// For now, this is fired on `scrollstart` and `scrollend`
+- (void)fireScrollEvent:(NSString*)eventName
+{
+    if([[self proxy] _hasListeners:eventName])
+    {
+        UITableView* tableView = [self tableView];
+        
+        NSIndexPath* indexPath = [self firstVisibleIndexPath:0];
```

Comment:
I would not do this at all. Just rely on the system supplied `indexPathsForVisibleRows` for the tableView. Less computation and fewer errors